examples/OneDShallowWaterGeostrophicAdjustment.jl

```diff
@@ -0,0 +1,197 @@
+# This example solves the 1D rotating shallow water equations
+# for the depth-integrated velocities u and v and for the
+# surface elevation η in a fluid with constant depth h,
+# gravitational acceleration g and Coriolis parameter f:
+#
+# ∂u/∂t + u∂u/∂x - fv = -gh∂η/∂x - D,   (1)
+# ∂v/∂t + u∂v/∂x + fu = - D,            (2)
+# ∂η/∂t + ∂u/∂x = - D.                  (3)
+#
+# where D indicates a hyperviscous operator of
+# the form ν*Laplacian^nν, where ν is the viscosity
+# coefficient and nν is the order of the operator.
+# The code time-steps Equations (1-3) simultaneously.
+#
+# The initial conditions prescribe a Gaussian bump larger
+# than the deformation radius in a fluid at rest, with
+# some random noise smaller than the deformation radius
+# added on top. The system develops geostrophically-balanced
+# jets around the Gaussian bump, while the smaller-scale noise
+# propagates around as inertia-gravity waves.
+
+using FourierFlows, PyPlot
+using FFTW: rfft, irfft, fft, ifft
+using LinearAlgebra: mul!, ldiv!
+using Printf
+using Random
+
+nothingfunction(args...) = nothing
+
+function calcN!(N, sol, t, s, v, p, g)
+  mul!(v.uh, g.rfftplan, deepcopy(v.u))
+  mul!(v.vh, g.rfftplan, deepcopy(v.v))
+  mul!(v.ηh, g.rfftplan, deepcopy(v.η))
+  ldiv!(v.ux, g.rfftplan, -im*g.kr.*v.uh)
+  ldiv!(v.vx, g.rfftplan, -im*g.kr.*v.vh)
+  mul!(v.uuxh, g.rfftplan, v.u.*v.ux)
+  mul!(v.uvxh, g.rfftplan, v.u.*v.vx)
+  rhsu = p.f*v.vh + p.g*p.h*im*g.kr.*v.ηh - v.uuxh
+  rhsv = - p.f*v.uh - v.uvxh
+  rhsη = im*g.kr.*v.uh
+  N[:, 1] .= rhsu
+  N[:, 2] .= rhsv
+  N[:, 3] .= rhsη
+  dealias!(N, g, g.kralias)
+  nothing
+end
+
+struct Vars{Aphys, Atrans} <: AbstractVars
+        u :: Aphys
+        v :: Aphys
+        η :: Aphys
+       ux :: Aphys
+       vx :: Aphys
+       uh :: Atrans
+       vh :: Atrans
+       ηh :: Atrans
+     uuxh :: Atrans
+     uvxh :: Atrans
+end
+
+function Vars(::Dev, g::AbstractGrid{T}) where {Dev, T}
+  @devzeros Dev T g.nx u v η ux vx
+  @devzeros Dev Complex{T} (g.nkr) uh vh ηh uuxh uvxh
+  mul!(uh, g.rfftplan, u)
+  mul!(vh, g.rfftplan, v)
+  mul!(ηh, g.rfftplan, η)
+  Vars(u, v, η, ux, vx, uh, vh, ηh, uuxh, uvxh)
+end
+
+struct Params{T} <: AbstractParams
+       ν :: T         # Hyperviscosity coefficient
+      nν :: Int       # Order of the hyperviscous operator
+      g  :: T         # Gravitational acceleration
+      h  :: T         # Fluid depth
+      f  :: T         # Coriolis parameter
+end
+Params(ν, nν, g, h, f) = Params(ν, nν, g, h, f)
+
+function Equation(p, g)
```

Comment:
```suggestion
function Equation(p, g::AbstractGrid{T}) where T
  LC = zeros(T, g.nkr, 3)
```
This way Equation will just use the same number type as that used in the Grid.